go/chat/attachment_crypt.go

```diff
@@ -1,45 +1,112 @@
 package chat
 
-import "io"
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+
+	"github.com/agl/ed25519"
+	"github.com/keybase/client/go/chat/signencrypt"
+	"github.com/keybase/client/go/libkb"
+)
 
 type Encrypter interface {
 	// EncryptedLen returns the number of bytes that the ciphertext of
 	// size plaintext bytes will be.
 	EncryptedLen(size int) int
 
 	// Encrypt takes a plaintext reader and returns a ciphertext reader.
-	Encrypt(plaintext io.Reader) (ciphertext io.Reader)
+	// It generates new keys every time it is called.
+	Encrypt(plaintext io.Reader) (ciphertext io.Reader, err error)
+
+	// EncryptKey returns the ephemeral key that was used during Encrypt.
+	EncryptKey() []byte
 
-	// Key returns the ephemeral key that was used during Encrypt.
-	Key() []byte
+	// VerifyKey returns the public portion of the signing key that
+	// can be used for signature verification.
+	VerifyKey() []byte
 }
 
 type Decrypter interface {
-	// Decrypt takes a ciphertext reader and a key and returns a
-	// plaintext reader.
-	Decrypt(ciphertext io.Reader, key []byte) (plaintext io.Reader, err error)
+	// Decrypt takes a ciphertext reader, encryption and verify keys.
+	// It returns a plaintext reader.
+	Decrypt(ciphertext io.Reader, encKey, verifyKey []byte) (plaintext io.Reader)
 }
 
-// PassThrough implements Encrypter and Decrypter.  It does no encryption
-// or decryption.
-type PassThrough struct{}
+var nonce signencrypt.Nonce
 
-func NewPassThrough() *PassThrough {
-	return &PassThrough{}
+func init() {
+	var n [signencrypt.NonceSize]byte
+	copy(n[:], "kbchatattachment")
+	nonce = &n
```

Comment:
If you wanted to avoid this static initialization stuff you could do

    // The ASCII bytes "kbchatattachment".
    var nonce signencrypt.Nonce = [16]byte{0x6b, 0x62, 0x63, 0x68, 0x61, 0x74, 0x61, 0x74, 0x74, 0x61, 0x63, 0x68, 0x6d, 0x65, 0x6e, 0x74}

Sucks that these are the only options. I'm fine either way.